test/Annotation/AbstractBuilderTestCase.php

```diff
@@ -261,7 +263,44 @@ public function testAllowsComposingMultipleChildEntities(): void
         self::assertTrue($target->has('username'));
         self::assertTrue($target->has('password'));
         self::assertInstanceOf(InputFilterProviderFieldset::class, $target);
-        $filterSpec = $target->getInputFilterSpecification();
+        $this->validateEntityFilterSpec($target->getInputFilterSpecification());
+    }
+
+    public function testAllowsComposingMultipleChildEntitiesWithEntityBind(): void
+    {
+        $entity  = new TestAsset\Annotation\EntityComposingMultipleEntitiesObjectPropertyHydrator();
+        $builder = $this->createBuilder();
+        $form    = $builder->createForm($entity);
+
+        self::assertTrue($form->has('child'));
+        $child = $form->get('child');
+        self::assertInstanceOf(Collection::class, $child);
+        $target = $child->getTargetElement();
+        self::assertInstanceOf(FieldsetInterface::class, $target);
+        self::assertTrue($target->has('username'));
+        self::assertTrue($target->has('password'));
+        self::assertInstanceOf(InputFilterProviderFieldset::class, $target);
+        $this->validateEntityFilterSpec($target->getInputFilterSpecification());
+
+        self::assertNull($entity->child);
+        $form->bind($entity);
+        $form->setData([
+            'child' => [
+                '0' => ['password' => 'email@test.com', 'username' => 'user'],
+                '1' => ['password' => 'email2@test.com', 'username' => 'user2'],
+            ],
+        ]);
+        self::assertTrue($form->isValid());
+        self::assertNotNull($entity->child);
```

Comment:
This PSalm `DocblockTypeContradiction` can be added to the baseline:
```console
vendor/bin/psalm --no-cache --set-baseline=psalm-baseline.xml --update-baseline
```

Reply:
File updated, thanks!